src/snowflake/snowpark/functions.py

```diff
@@ -465,6 +473,40 @@ def bitshiftright(to_shift_column: ColumnOrName, n: Union[Column, int]) -> Colum
     return call_builtin("bitshiftright", c, n)
 
 
+def bround(col: ColumnOrName, scale: Union[Column, int]) -> Column:
+    """
+    Rounds the number using `HALF_TO_EVEN` option. The `HALF_TO_EVEN` rounding mode rounds the given decimal value to the specified scale (number of decimal places) as follows:
+    * If scale is greater than or equal to 0, round to the specified number of decimal places using half-even rounding. This rounds towards the nearest value with ties (equally close values) rounding to the nearest even digit.
+    * If scale is less than 0, round to the integral part of the decimal. This rounds towards 0 and truncates the decimal places.
+    NOTE: Values are cast to NUMBER(20,8) prior rounding.
+
+    Example:
+        >>> data = [(1.235),(3.5)]
+        >>> df = session.createDataFrame(data, ["value"])
+        >>> df.select(bround('VALUE',1).alias("VALUE")).show() # Rounds to 1 decimal place
+        -----------
+        |"VALUE"  |
+        -----------
+        |1.2      |
+        |3.5      |
+        -----------
+        <BLANKLINE>
+        >>> df.select(bround('VALUE',0).alias("VALUE")).show() # Rounds to 1 decimal place
+        -----------
+        |"VALUE"  |
+        -----------
+        |1        |
+        |4        |
+        -----------
+        <BLANKLINE>
+    """
+    col = _to_col_if_str(col, "bround")
+    scale = _to_col_if_lit(scale, "bround")
+    return call_builtin(
+        "ROUND", col.cast(DecimalType(20, 8)), scale, lit("HALF_TO_EVEN")
```

Comment:
@sfc-gh-mrojas As discussed offline, we're not going to explicitly cast to a certain precession and scale in the SDK. We give the control to users.